src/Inkluzitron/Modules/Vote/VotegagReactionHandler.cs

```diff
@@ -0,0 +1,151 @@
+        private IMemoryCache Cache { get; }
+
+        static private readonly Regex VoteMessageMatch
+            = new(@"^Hlasování o umlčení.*?(\S+) na dobu \*\*(\d+).*?alespoň (\d+).*?Hlasování bude", RegexOptions.Singleline);
+
+        public VotegagReactionHandler(UsersService usersService, VoteSettings voteSettings,
+            DiscordSocketClient client, IMemoryCache cache, BotSettings botSettings)
+        {
+            UsersService = usersService;
+            VoteSettings = voteSettings;
+            Client = client;
+            Cache = cache;
+            BotSettings = botSettings;
+
+            client.Ready += ClientReadyAsync;
+            client.ChannelCreated += UpdateChannelMutedRoleAsync;
+        }
+
+        private async Task UpdateChannelMutedRoleAsync(SocketChannel channel)
+        {
+            if (channel is not SocketGuildChannel guildChannel) return;
+
+            var muteRole = guildChannel.Guild.GetRole(VoteSettings.MuteRoleId);
+
+            var isPermSet = guildChannel.PermissionOverwrites.Any(o =>
+                 o.TargetType == PermissionTarget.Role &&
+                 o.TargetId == muteRole.Id &&
+                 o.Permissions.SendMessages == PermValue.Deny);
+
+            if (isPermSet) return;
+
+            var rolePerms = new OverwritePermissions(sendMessages: PermValue.Deny);
+
+            await guildChannel.AddPermissionOverwriteAsync(muteRole, rolePerms);
+            
+        }
+
+        private async Task ClientReadyAsync()
+        {
+            var guild = Client.GetGuild(BotSettings.HomeGuildId);
+            var muteRole = guild.GetRole(VoteSettings.MuteRoleId);
+
+            await SetupMutedRoleAsync(guild);
+
+            foreach (var user in guild.Users)
+            {
+                var userDb = await UsersService.GetUserDbEntityAsync(user);
+
+                if (userDb == null || userDb.MutedUntil == null)
+                {
+                    if(user.Roles.Any(r => r.Id == muteRole.Id))
+                        await user.RemoveRoleAsync(muteRole);
+                    continue;
+                }
+
+                if (userDb.MutedUntil <= DateTime.Now)
+                {
+                    await user.RemoveRoleAsync(muteRole);
+                    await UsersService.SetMutedUntilAsync(user, null);
+                    continue;
+                }
+
+                ScheduleUnmute(user, (DateTime)userDb.MutedUntil, muteRole);
+            }
+        }
+
+        private async Task SetupMutedRoleAsync(SocketGuild guild)
+        {
+            foreach (var channel in guild.Channels)
+            {
+                await UpdateChannelMutedRoleAsync(channel);
+            }
+        }
+
+        private void ScheduleUnmute(SocketGuildUser user, DateTime endTime, SocketRole mutedRole)
+        {
+            Task.Delay(endTime - DateTime.Now).ContinueWith(_ =>
+                VotegagModule.UnmuteUserAsync(user, UsersService, mutedRole));
```

Comment:
když to máš takhle v ContinueWith, tak se nikdy nedozvíš, jak to dopadlo, když dojde k výjimce tak se ti to spolkne
minimálně bych přidal continuation, co se spustí když TaskContinuationOptions.OnlyOnFaulted, aby se to aspoň zalogovalo